### ISSUE TITLE
[Xliff] Allow "approving" translations

### ISSUE THREAD

**Issue by rvanlaak** [php-translation/symfony-bundle#64]
When translations are new, or when they have been edited lately, a workflow to approve them would help to ensure translation quality. The Xliff  standard does support this out of the box, the bundle can help with that. Also see [`XliffExporter`](https://github.com/lexik/LexikTranslationBundle/blob/master/Translation/Exporter/XliffExporter.php).

**Scenario 1: new translation**

* developer introduces a new translation key in a Twig view.
* translation bundle collects this key, and adds it to the `xliff` file with [`approved` attribute set to `no`](http://docs.oasis-open.org/xliff/v1.2/os/xliff-core.html#approved) if the translation has a value.
* Web UI has specific styling / workflow for the non-approved translations, translator goes to UI to review those specific translations.
* The "approve translation" action sets `approved` to `yes`

**Scenario 2: updating translation via edit in place or profiler**

* translator uses edit in place or profiler to edit a translation
* POST action sets `approved` back to `no`
* same "approve" workflow via Web UI as in scenario 1

**Scenario 3: updating translation web ui**

* translator uses web ui to translate a key
* there are two "save" buttons, the normal "save" one (behaves as scenario 2), and a "save + approve" button
* translator saves and approves a translation key

Future enhancement could be to require a certain `ROLE` in order to approve a translation.

**Comment by Nyholm** [comment]
I like the ability of a support function. This is however only a xliff feature. But xliff is the standard =)

👍 

To implement this we need to make changes to the CatalogueManager and the CatalougeMessage.

**Comment by rvanlaak** [comment]
If we can agree on the technical specs / changes, I'd like to help with the implementation right away 👍 

One UI thing I mentioned already is that the web UI uses colors, but I think they are related to the completion percentage (?). These colors should change: gray or red is missing, green is approved, yellow is not approved yet.

**Comment by Nyholm** [comment]
Make many smaller PRs, they are easier to review and agree upon. Start with just adding this support to the CatalogueManager end CatalogueMessage. 

Then you may make two more PRs, one for WebUI and one for EditInPlace.

**Comment by rvanlaak** [comment]
I first need to finish our migration to `xliff`, so I can probably start working on this in the beginning of next week.

**Comment by Nyholm** [comment]
FYI. Using symfony-storage 0.3.0 will allow storing "approved".

**Comment by rvanlaak** [comment]
@Nyholm any status update on the symfony-storage? Did it get any additional features that would be able to help us even better while implementing this?

**Comment by Nyholm** [comment]
Yeah. You've probably seen it already. But there is a `Metadata` class. If you use XLIFF then you will be able to save the metadata.
